### PR TITLE
Feature/gs form alterations

### PR DESCRIPTION
**Pull request by martinyde**
https://jira.itkdev.dk/browse/JE-186

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -81,5 +81,12 @@ SITE_NAME="Jira economics"
...
-MAILER_NAME="Jira economics"
-MAILER_EMAIL="info@jira-economics.com"
+
+###> symfony/swiftmailer-bundle ###
+# For Gmail as a transport, use: "gmail://username:password@localhost"
+# For a generic SMTP server, use: "smtp://localhost:25?encryption=&auth_mode="
+# Delivery is disabled by default via "null://localhost"
+MAILER_NAME="ITK"
+MAILER_URL=smtp://mailhog:1025
+MAILER_EMAIL="info@support.srvitkphp72stg.itkdev.dk"
```
Comment (tuj) [inline .env]: Doesn't this belong in .env.local ?

File: bundles/GraphicServiceOrder/Service/OrderService.php [hunk 2]
```diff
@@ -74,6 +85,9 @@ public function __construct(
         $this->tokenStorage = $tokenStorage;
         $this->fileUploader = $fileUploader;
         $this->userManager = $userManager;
+        $this->swiftMailer = $swiftMailer;
+        $this->twig = $twig;
+        $this->translator = $translator;
```
Comment (tuj) [inline bundles/GraphicServiceOrder/Service/OrderService.php:105]: Missing in top of class.

File: bundles/GraphicServiceOrder/Service/OrderService.php [hunk 3]
```diff
@@ -219,6 +251,10 @@ private function createOrderTask(GsOrder $gsOrder)
                 'issuetype' => [
                     'id' => $this->params->get('gs_order_issuetype_id'),
                 ],
+                'reporter' => [
+                    'name' => $author,
+                ],
+                'customfield_10309' => (string) $gsOrder->getDebitor(),
```
Comment (tuj) [inline bundles/GraphicServiceOrder/Service/OrderService.php]: Custom field ids can change between instances.
See https://github.com/aakb/jira_economics/blob/develop/bundles/Billing/Service/BillingService.php#L806
for an example of how to get the id of the custom field.

File: bundles/GraphicServiceOrder/Service/OrderService.php [hunk 4]
```diff
@@ -322,12 +358,39 @@ private function storeFile(GsOrder $gsOrder, $form)
...
         }
         $gsOrder->setFiles($uploadedFiles);
 
         return $gsOrder;
     }
+
+    /**
+     * @param $name
+     *
+     * Send receipt mail
```
Comment (tuj) [inline bundles/GraphicServiceOrder/Service/OrderService.php]: Description should be at the top.

File: .env (hunk 1 above)
Comment (tuj) [reply .env]: If you want values that applies to dev environment use .env.dev

**Review by tuj — CHANGES_REQUESTED** [review]

File: bundles/GraphicServiceOrder/Service/OrderService.php (hunk 2 above)
Comment (martinyde) [reply bundles/GraphicServiceOrder/Service/OrderService.php:105]: Please explain, "top of class", looks good to me.

File: bundles/GraphicServiceOrder/Service/OrderService.php (hunk 3 above)
Comment (martinyde) [reply bundles/GraphicServiceOrder/Service/OrderService.php]: Added the field id .env instead. I believe it's a better approach.

File: .env (hunk 1 above)
Comment (martinyde) [reply .env]: Replaced by empty strings

File: .env [hunk 5]
```diff
@@ -86,7 +90,7 @@ SITE_LOGO_PATH="build/images/itk-jira-symbol.svg"
 # For Gmail as a transport, use: "gmail://username:password@localhost"
 # For a generic SMTP server, use: "smtp://localhost:25?encryption=&auth_mode="
 # Delivery is disabled by default via "null://localhost"
-MAILER_NAME="ITK"
-MAILER_URL=smtp://mailhog:1025
-MAILER_EMAIL="info@support.srvitkphp72stg.itkdev.dk"
+MAILER_NAME=""
```
Comment (rimi-itk) [inline .env]: This should be set to a reasonable default, e.g `JIRA Economics`, to make it possible to run the project without adding local configuration in `.env.local`.

File: .env [hunk 6]
```diff
@@ -86,7 +90,7 @@ SITE_LOGO_PATH="build/images/itk-jira-symbol.svg"
...
+MAILER_URL=
```
Comment (rimi-itk) [inline .env]: This should be set to `smtp://mailhog:1025`.

File: .env [hunk 7]
```diff
@@ -86,7 +90,7 @@ SITE_LOGO_PATH="build/images/itk-jira-symbol.svg"
...
+MAILER_EMAIL=""
```
Comment (rimi-itk) [inline .env]: Set this to some example email address, e.g. `info@jira_economics.docker.localhost`.

File: .env [hunk 8]
```diff
@@ -86,7 +90,7 @@ SITE_LOGO_PATH="build/images/itk-jira-symbol.svg"
...
 ###< symfony/swiftmailer-bundle ###
```
Comment (rimi-itk) [inline .env]: Missing newline.

File: bundles/GraphicServiceOrder/Service/OrderService.php [hunk 9]
```diff
@@ -282,7 +282,7 @@ private function createFolder($order_key)
     }
 
     /**
-     * Share file in owncloud.
+     * Share file in Owncloud.
```
Comment (rimi-itk) [inline bundles/GraphicServiceOrder/Service/OrderService.php]: It's actually called “ownCloud” (cf. https://owncloud.org/trademark/).

**Review by rimi-itk — CHANGES_REQUESTED** [review]
A few comments added.

File: bundles/GraphicServiceOrder/Service/OrderService.php (hunk 4 above)
Comment (martinyde) [reply bundles/GraphicServiceOrder/Service/OrderService.php]: fixed

File: bundles/GraphicServiceOrder/Service/OrderService.php (hunk 9 above)
Comment (martinyde) [reply bundles/GraphicServiceOrder/Service/OrderService.php]: Fixed

File: .env (hunk 7 above)
Comment (martinyde) [reply .env]: changed back

File: .env (hunk 6 above)
Comment (martinyde) [reply .env]: changed back

File: .env (hunk 8 above)
Comment (martinyde) [reply .env]: Fixed

File: .env (hunk 5 above)
Comment (martinyde) [reply .env]: changed back

File: bundles/GraphicServiceOrder/Service/OrderService.php (hunk 2 above)
Comment (tuj) [reply bundles/GraphicServiceOrder/Service/OrderService.php:105]: $this->translator is not declared at the top of the class.

`private $translator;`

Comment (martinyde) [reply bundles/GraphicServiceOrder/Service/OrderService.php:105]: Ahh, true

**Review by rimi-itk — APPROVED** [review]
Godkendt!

**Review by tuj — APPROVED** [review]

**Pull request merged**